staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/plugin_test.go

```diff
@@ -251,3 +252,80 @@ func TestAdmitCachedClient(t *testing.T) {
 		}
 	}
 }
+
+// TestWebhookDuration tests that MutatingWebhook#Admit sets webhook duration in context correctly
+func TestWebhookDuration(t *testing.T) {
```

Comment:
I think we want a bit better test at least exercises existence of multiple webhooks and that it not only checks if duration is zero, but does something a bit more sophisticated.

Here is how I would do this:
1) we add a clock field to Plugin (and we initialize it in NewMutatingWebhook to RealClock)
2) we use a clock to get time and compute difference in Admit()

3) here in this test, for webhooks, we explicitly set their clock field to FakeClock
4) we actually register multiple webhooks
     (let's not reuse the NewNonMutatingTestCases, but create your own set of webhooks probably)
5) add NewTestServerWithHandler (next to NewTestServer function) that will allow you to inject your handler
6) create your handler that will basically be mving the clock, e.g. your handler will be something like:
```
func webhookHandler(w http.ResponseWriter, r. *http.Request) {
  switch r.URL.Path {
  case "/sleep100":
    c.Step(100*time.Millsecond) 
    { the same stuff that is returned in the original handler for /allow path}
  case "/sleep500":
   c.Step(500*time.Millisecond)
   { the same stuff that is returned in the original handler for /allow path}
  default:
    http.NotFound(w, r)
  }
}
```

7) in the test, validate that the clock was actually moved by expected count (e.g. if you registered one webhook for /sleep100 and one for /sleep500, you wan't to validate that reported admit time was actually 600ms

Reply:
i think unit tests would become more simple and concise with this abstraction - https://github.com/kubernetes/kubernetes/pull/105890#discussion_r742863174